### PR TITLE
Multiple MIG feature added

### PR DESCRIPTION
**Pull request by JBleher**
**Summary of changes:**
- Added the possibility to use several MIGs with one pod.
**Checklist:**
I don't know what to change in the _Chart.yml_ for artifacthub.io/changes ??
* [ ] I updated the `artifacthub.io/changes` in _Chart.yml_

### CODE REVIEW

**Comment by jdetroyes** [comment]
Hi @JBleher 

We will take your request soon for MIG!

File: templates/deployment.yaml [hunk 1]
```diff
@@ -153,13 +169,25 @@ spec:
...
+              ( or (eq .Values.ollama.gpu.type "nvidia") (not .Values.ollama.gpu.type)) 
+              (( .Values.ollama.gpu.mig.enabled))
+            ) }}
+            {{- range $key, $value := .Values.ollama.gpu.mig.devices }}
+        - key: nvidia.com/mig-{{ $key }}
+          operator: Exists
+          effect: NoSchedule
+            {{- end }}
+          {{- end }}
         {{- end }}
```
Comment (jdetroyes) [inline templates/deployment.yaml:189]: It should only have two closing tags

File: templates/deployment.yaml [hunk 2]
```diff
@@ -153,13 +169,25 @@ spec:
...
         {{- with .Values.tolerations }}
           {{- toYaml . | nindent 8 }}
         {{- end }}
       {{- end }}
-{{- end }}
```
Comment (jdetroyes) [inline templates/deployment.yaml:165]: And this one should not be removed

**Review by jdetroyes — CHANGES_REQUESTED** [review]
Hello @JBleher 

Can you check these changes.

File: templates/deployment.yaml (hunk 2 above)
Comment (JBleher) [reply templates/deployment.yaml:165]: Thank you yes. Good catch. I will.

**Comment by JBleher** [comment]
@jdetroyes corrected the template. Also changed the inserted part in the values to an empty list, since that caued problems in my particular implementation.

**Comment by JBleher** [comment]
Also: I don't understand the error code in the CI pipeline? Sorry... would fix it, but I don't know what causes it...

------------------------------------------------------------------------------------------------------------------------

 ✖︎ ollama => (version: "0.53.0", path: ".") > failed waiting for process: exit status 1

------------------------------------------------------------------------------------------------------------------------

failed linting and installing charts: failed processing charts

Error: failed linting and installing charts: failed processing charts

File: values.yaml [hunk 3]
```diff
@@ -51,11 +51,22 @@ ollama:
...
     nvidiaResource: "nvidia.com/gpu"
     # nvidiaResource: "nvidia.com/mig-1g.10gb" # example
+    # If you want to use more than one NVIDIA MIG you can use the following syntax (then nvidiaResource is ignored and only the configuration in the following MIG section is used)
+
+    mig:
+      # If enabled you will have to specify the mig devices
+      # If enabled is set to false this section is ignored
+      enabled: false
+      # Specify the mig devices and the corresponding number
+      devices: []
```
Comment (jdetroyes) [inline values.yaml:67]: It should be `{}` as it's a map not an array.

**Review by jdetroyes — CHANGES_REQUESTED** [review]
Hey @JBleher 

One last thing before we merge, can you please check the default value change I mentioned.

**Review by jdetroyes — APPROVED** [review]

**Pull request merged**